dali/test/python/test_dali_tf_dataset_pipelines.py

```diff
@@ -0,0 +1,271 @@
+# Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import nvidia.dali as dali
+import nvidia.dali.fn as fn
+import nvidia.dali.plugin.tf as dali_tf
+from nvidia.dali import pipeline_def
+
+
+import tensorflow as tf
+from test_utils import  RandomlyShapedDataIterator
+
+import numpy as np
+
+
+from nose.tools import nottest
+
+class RandomSampleIterator:
+    def __init__(self,
+                 max_shape=(10, 600, 800, 3),
+                 dtype_sample=np.uint8(0),
+                 start=0,
+                 stop=10e100,
+                 min_shape=None,
+                 seed=42):
+        self.start = start
+        self.stop = stop
+        self.min_shape = min_shape
+        self.max_shape = max_shape
+        # As tf passes only tensors to the iterator, we pass a dummy value of which we take the type
+        self.dtype = dtype_sample.dtype
+        self.seed = seed
+
+    def __iter__(self):
+        self.n = self.start
+        self.random_iter = iter(RandomlyShapedDataIterator(batch_size=1, min_shape=self.min_shape,
+                max_shape=self.max_shape, seed=self.seed, dtype=self.dtype))
+        return self
+
+    def __next__(self):
+        if self.n <= self.stop:
+            result = self.n
+            self.n += 1
+            ret = self.random_iter.next()[0]
+            return ret
+        else:
+            raise StopIteration
+
+
+class FixedSampleIterator:
+    def __init__(self, value):
+        self.value = value
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        return self.value
+
+class InfiniteSampleIterator:
+    def __init__(self, start_value):
+        self.value = start_value
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        result = self.value
+        self.value = self.value + np.array(1, dtype=self.value.dtype)
```

Comment:
```suggestion
        self.value = self.value + np.array(1, dtype=self.value.dtype)
```
Why that expression? Is it for the sake of generating an array?

Reply:
It is for sake of keeping the numpy from making unexpected type promotion.

Reply:
```suggestion
        self.value = self.value + np.ones_like(self.value)
```
?